blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/JSONMetricsContainer.java

```diff
@@ -142,6 +147,13 @@ public void setTtlInSeconds(int ttlInSeconds) {
             this.ttlInSeconds = ttlInSeconds;
         }
 
+        public boolean isValid() {
+            long current = System.currentTimeMillis();
+
+            // valid if collectionTime is greater than past window or less than future window
+            return ((collectionTime >= current - pastDiff ) && ( collectionTime <= current + futureDiff ));
```

Comment:
What if we instead add an "ingestTime" property at the handler level which retrieve the System.currentTimeMillis() value once when the request handling started, and use that value for all basis of validity, delays, etc?  We will have to pass this value to the validation and delay tracking/segregation code to use instead of having them call System.currentTimeMillis() when needed.

Let me think about this a little more.